DSCResources/MSFT_xDnsRecord/MSFT_xDnsRecord.psm1

```diff
@@ -46,12 +60,25 @@ function Set-TargetResource
         $Zone,
 
         [parameter(Mandatory = $true)]
+        [ValidateSet("A-record", "C-name")]
+        [System.String]
+        $Type,
+
+        [parameter(Mandatory = $true)]
         [System.String]
         $Target
     )
-    
-    Write-Verbose "Creating for DNS $Target in $Zone"
-    Add-DnsServerResourceRecordA -IPv4Address $Target -Name $Name -ZoneName $Zone -ComputerName "localhost" 
+
+    if ($Type -eq "A-record")
+    {
+        Write-Verbose "Creating $Type for DNS $Target in $Zone"
+        Add-DnsServerResourceRecordA -IPv4Address $Target -Name $Name -ZoneName $Zone
```

Comment:
Would it be better to use [Add-DnsServerResourceRecord](https://technet.microsoft.com/en-us/library/jj649925.aspx) cmdlet instead along with splatting so adding more types later will not bloat the code. Something like
```powershell
$parameters = @{IPv4Address = $Target; Name=$Name; ZoneName=$Zone} 
```
and in the if/else decide to add -A or -CName switch parameters as
```powershell 
if($type -eq 'CName'){$parameters.Add('CNAME',$true) } 
```
and outside of if/else splat to [Add-DnsServerResourceRecord](https://technet.microsoft.com/en-us/library/jj649925.aspx)  cmdlet as 
```powershell 
Add-DnsServerResourceRecord @parameters
```